internal/auth/options.go

```diff
@@ -260,7 +261,14 @@ func newProvider(o *Options) (providers.Provider, error) {
 
 	var singleFlightProvider providers.Provider
 	switch o.Provider {
-	case providers.GoogleProviderName: // Google
+	case providers.AzureProviderName:
+		azureProvider, err := providers.NewAzureV2Provider(p)
+		if err != nil {
+			return nil, err
+		}
+		azureProvider.Configure(o.AzureTenant)
+		singleFlightProvider = providers.NewSingleFlightProvider(azureProvider)
+	case providers.GoogleProviderName:
```

Comment:
Optional for wiring up OIDC as a provider:

```suggestion
	case providers.OIDCProviderName:
		oidcProvider, err := providers.NewOIDCProvider(p, o.OIDCDiscoveryURL)
		if err != nil {
			return nil, err
		}
		singleFlightProvider = providers.NewSingleFlightProvider(oidcProvider)
	case providers.GoogleProviderName:
```